server/application/rest/service_commitment.py

```diff
@@ -0,0 +1,58 @@
+"""
+Module for handling service commitment API endpoints.
+Provides endpoints for creating and retrieving service commitments.
+"""
+from flask import Blueprint, request, jsonify
+from application.rest.work_shift import get_user_from_token, HTTP_STATUS_CODES_MAPPING, ResponseTypes
+from use_cases.add_service_commitments import add_service_commitments, get_service_commitments
+from repository.mongodb.service_commitments import MongoRepoCommitments  # Adjust the import path as needed
+
+blueprint = Blueprint("service_commitment", __name__)
+
+repo = MongoRepoCommitments()
+
+@blueprint.route("/service_commitment", methods=["POST"])
+def create_service_commitment():
+    """
+    Handle POST request to create service commitments.
+    Extract user info from Authorization token and create commitments.
+    """
+    try:
+        user_email = get_user_from_token(request.headers)
```

Comment:
get_user_from token returns a dictionary with 'email', 'firstName', and 'lastName' as keys. You should extract the email only. Right now, user_email is getting saved in the database as:
```
user_email
{
"developer@slu.edu",
"SLU",
"Developer"
}
```